File: Interview Preparation Kit - Python/03. Dictionaries and Hashmaps/002. Two Strings.py
# Problem: https://www.hackerrank.com/challenges/two-strings/problem
#  Score: 25


for i in range(int(input())):
    str1 = set(input())
    str2 = set(input())
    if str1 & str2 != set():
        print('YES')
    else:
        print('NO')

File: Interview Preparation Kit - CPP/03. Dictionaries and Hashmaps/002. Two Strings.cpp
// Problem: https://www.hackerrank.com/challenges/two-strings/problem
// Score: 25


#include <iostream>
#include <set>
#include <string>
using namespace std;


int main(){
    int p;
    cin >> p;

    for (int i = 0; i < p; i++){
        string s1, s2;
        cin >> s1 >> s2;

        set<char> letters;

        for (char c: s1){
            letters.insert(c);
        }

        bool ans = false;
        for (char c: s2){
            if (letters.find(c) != letters.end()){
                ans = true;
                break;
            }
        }

        if (ans) cout << "YES" << endl;
        else cout << "NO" << endl;
    }

    return 0;
}
